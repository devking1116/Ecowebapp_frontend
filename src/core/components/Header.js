/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Image } from 'semantic-ui-react';

import FillRemaining from './FillRemaining';

type Props = {
  logoImgSrc: string,
  children?: any,
};

/**
 * The header for the website.
 */
export default function Header({ logoImgSrc, children, ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      header: {
        height: 40,

        //TODO, move to theme
        backgroundColor: '#468dc8',
      },
      logo: {
        height: 40,
        padding: 2,
      },
      children: {
        display: 'flex',
        justifyContent: 'space-between',
        height: 40,
      },
    },
  });

  return (
    <header style={styles.header} {...otherProps}>
      <FillRemaining direction="row">
        <Image style={styles.logo} src={logoImgSrc} size="tiny" />
        <div style={styles.children}>
          {children}
        </div>
      </FillRemaining>
    </header>
  );
}
