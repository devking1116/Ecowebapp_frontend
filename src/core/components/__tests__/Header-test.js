import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../Header';

test('it displays the logo using given url.', () => {
  const component = renderer.create(
    <Header logoImgSrc="/path/to/logo.png" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it displays the children on the right side.', () => {
  const component = renderer.create(
    <Header>
      Header test children
    </Header>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
