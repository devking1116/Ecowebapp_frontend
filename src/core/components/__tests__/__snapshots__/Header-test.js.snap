exports[`test it displays the children on the right side. 1`] = `
<header
  style={
    Object {
      "backgroundColor": "#468dc8",
      "height": 40,
    }
  }>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "100%",
        "width": "100%",
      }
    }>
    <div
      style={
        Object {
          "height": "100%",
        }
      }>
      <img
        alt={undefined}
        className="ui tiny image"
        height={undefined}
        src={undefined}
        style={
          Object {
            "height": 40,
            "padding": 2,
          }
        }
        width={undefined} />
    </div>
    <div
      style={
        Object {
          "MozBoxFlex": "1 0 1px",
          "WebkitBoxFlex": "1 0 1px",
          "WebkitFlex": "1 0 1px",
          "flex": "1 0 1px",
          "height": "100%",
          "msFlex": "1 0 1px",
        }
      }>
      <div
        style={
          Object {
            "WebkitJustifyContent": "space-between",
            "display": "flex",
            "height": 40,
            "justifyContent": "space-between",
          }
        }>
        Header test children
      </div>
    </div>
  </div>
</header>
`;

exports[`test it displays the logo using given url. 1`] = `
<header
  style={
    Object {
      "backgroundColor": "#468dc8",
      "height": 40,
    }
  }>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "100%",
        "width": "100%",
      }
    }>
    <div
      style={
        Object {
          "height": "100%",
        }
      }>
      <img
        alt={undefined}
        className="ui tiny image"
        height={undefined}
        src="/path/to/logo.png"
        style={
          Object {
            "height": 40,
            "padding": 2,
          }
        }
        width={undefined} />
    </div>
    <div
      style={
        Object {
          "MozBoxFlex": "1 0 1px",
          "WebkitBoxFlex": "1 0 1px",
          "WebkitFlex": "1 0 1px",
          "flex": "1 0 1px",
          "height": "100%",
          "msFlex": "1 0 1px",
        }
      }>
      <div
        style={
          Object {
            "WebkitJustifyContent": "space-between",
            "display": "flex",
            "height": 40,
            "justifyContent": "space-between",
          }
        } />
    </div>
  </div>
</header>
`;
